monailabel/tasks/infer/basic_infer.py

```diff
@@ -467,11 +467,10 @@ def _get_network(self, device, data):
                 if path:
                     checkpoint = torch.load(path, map_location=torch.device(device))
                     model_state_dict = checkpoint.get(self.model_state_dict, checkpoint)
-
-                    if set(self.network.state_dict().keys()) != set(checkpoint.keys()):
```

Comment:
does this change break any existing checkpoints for other models? bundles? other apps?

Reply:
I don't think so. Because of this line `model_state_dict = checkpoint.get(self.model_state_dict, checkpoint)`. If the checkpoint is the model_state_dict, then it will return the checkpoint directly. Otherwise it will return the model_state_dict. I think this is why this line exists here. And the old condition logic which uses the checkpoint looks like a bug to me. Please correct me if there are other cases.

Thanks  
Bin